### ISSUE TITLE
State value to set for "yamaha.0.Realtime.MAIN.VOL" has to be type "number" but received type "string" 

### ISSUE THREAD

**Issue by jack-blackson** [iobroker-community-adapters/ioBroker.yamaha#228]
When changing the volume, I receive these messages in the debug log:

<img width="1149" alt="grafik" src="https://user-images.githubusercontent.com/13975634/143784417-5d8b822b-29fc-4f2f-83ce-686bcbaef13f.png">

**Versions:**  
 - Adapter version: 0.4.0
 - JS-Controller version: 3.3.20
 - Node version: 12.16.3
 - Operating system: linux

**Comment by Sneak-L8** [comment]
I just checked the code. State is created as string. Would it be possible, that state were created by an older version of adapter? Try to delete state and restart adapter. Is problem solved or still there?

**Comment by jack-blackson** [comment]
Hey! I just gave it a try, deleted all objects, restarted the instance of the adapter, and unfortunately still get a ton of error messages: 
<img width="1293" alt="grafik" src="https://user-images.githubusercontent.com/13975634/157114102-053a728a-98e7-44a1-9fc4-ddfd5e96a646.png">

**Comment by Sneak-L8** [comment]
I think you're still using 0.4.0. Please update to 0.4.1 on git and test again.

**Comment by jack-blackson** [comment]
You are right. I ran the update to 0.4.1, just one info error when starting the adapter, and the volume one when changing volume: 
<img width="1146" alt="grafik" src="https://user-images.githubusercontent.com/13975634/157115071-a5c6856a-0487-416f-8c55-b89ae72b5b14.png">

**Comment by Sneak-L8** [comment]
Ok, I'll hve to check. Please give me some days ...

**Comment by Sneak-L8** [comment]
Can you check definition of state yamaha.0.Realtime.MAIN.VOL.
Here I see the following:
`{
  "type": "state",
  "common": {
    "name": "VOL",
    "type": "string",
    "role": "state"
  },
  "native": {},
  "_id": "yamaha.0.Realtime.MAIN.VOL",
  "from": "system.adapter.yamaha.0",
  "user": "system.user.admin",
  "ts": 1645102109067,
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}`
You'll find this in objects panel when clicking at the pencil at the right side of the state.

**Comment by jack-blackson** [comment]
Looks for me like that: {
  "type": "state",
  "common": {
    "name": "VOL",
    "type": "number",
    "role": "state"
  },
  "native": {},
  "_id": "yamaha.0.Realtime.MAIN.VOL",
  "from": "system.adapter.yamaha.0",
  "user": "system.user.admin",
  "ts": 1646685879012,
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}

**Comment by Sneak-L8** [comment]
Sorry, to ask again: you're sure that you deleted the whole state, not only the value of the state?

**Comment by jack-blackson** [comment]
I don't know why, but I deleted the whole 0 folder, and now no error for the volume any more, and it is now also a string, like yours. Just the error for Pure Direct is left. 

info | State value to set for "yamaha.0.pureDirect" has to be type "boolean" but received type "string"

{
  "_id": "yamaha.0.pureDirect",
  "type": "state",
  "common": {
    "name": "pureDirect",
    "type": "boolean",
    "role": "state",
    "read": true,
    "write": true,
    "def": false,
    "desc": "Pure Direct"
  },
  "native": {},
  "from": "system.adapter.yamaha.0",
  "user": "system.user.admin",
  "ts": 1646853703480,
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}

**Comment by Sneak-L8** [comment]
It seems that indeed, pureDirect will be set to "On" and "Off" and not true and false. Therefore I changed definition of state. Please delete state pureDirect and update to latest version in git.

**Comment by jack-blackson** [comment]
I did the update, deleted all states, restarted the instance, and no error messages any more - good job, thanks for your work!!

**Comment by Sneak-L8** [comment]
You're welcome

**Comment by thebilberry** [comment]
Meanwhile I have updated to the latest Version (0.5.1) to solve the mentioned issue. But also after deleting states and later a complete reinstall of the adapter, the type is always set as "number". What am I doing wrong? Any hints?

**Comment by Sneak-L8** [comment]
Please show a screenshot of the state and of the settings (pencil icon)

**Comment by thebilberry** [comment]
{
  "type": "state",
  "common": {
    "name": "VOL",
    "type": "number",
    "role": "state"
  },
  "native": {},
  "_id": "yamaha.0.Realtime.MAIN.VOL",
  "from": "system.adapter.yamaha.0",
  "user": "system.user.admin",
  "ts": 1649000043080,
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}

**Comment by Sneak-L8** [comment]
What are the values stored in this state? Numbers or strings? Do you get warning messages about wrong types like the original poster?

**Comment by thebilberry** [comment]
yes, it's the same massage:
 `State value to set for "yamaha.0.Realtime.MAIN.VOL" has to be type "number" but received type "string"` 

That's why I can't explain it and asked. 
After installing the adapter, the state does not yet exist. Only after using functions - e.g. the volume buttons on the remote control - the state is (wrong) created again.

**Comment by thebilberry** [comment]
Maybe I found a workaround: I manually changed the state type of xxx.**MAIN.VOL** to "string". After that, the message was gone. (Let's see how long)
In this context, however, I have noticed three "new" similar messages:

State value to set for "**yamaha.0.Realtime.NETRADIO.CURRLINE**" has to be type "number" but received type "string"
State value to set for "**yamaha.0.Realtime.NETRADIO.MAXLINE**" has to be type "number" but received type "string"
State value to set for "**yamaha.0.Realtime.NETRADIO.LISTLAYER**" has to be type "number" but received type "string"

**Comment by Apollon77** [comment]
Reopen because? Please post new logs and such ith a current version

**Comment by Sneak-L8** [comment]
Reopen because he said, that thre are still messages about wrong type. I closed it after the first fix, but then additional issues were found. So I opened it again.

**Comment by Apollon77** [comment]
I would love to get a new issue with a current version and a new log :-)) because else we mix toooo much together
